File: python/detectcapital.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
       
        if not word:
            return False
        if word.isupper():
            return True
        if word.islower():
            return True
        if word[0].isupper():
            for i in xrange(1,len(word)):
                if word[i].isupper():
                    return False
            else:
                return True
        return False

File: cpp/detectcapital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int cnt = 0;
        for (int i = 0; i < word.length(); i++) {
            if (cnt > 1 and islower(word[i])) return false;
            if (isupper(word[i])) {
                cnt++;
            }
            
        }
        return cnt == word.length() || cnt == 0 || cnt == 1 && isupper(word[0]);
    }
};
